From a5018606c77ef32ac1df3412712804266016733d Mon Sep 17 00:00:00 2001
From: Jan Moeller <Jan.Moeller@governikus.de>
Date: Thu, 28 Aug 2025 11:31:22 +0200
Subject: a11y iOS: Inform VoiceOver about used language

The property accessibilityLanguage is used by VoiceOver on iOS and
allows to tell the screen reader which language package to use. This
is helpful if the app's language is different from the OS language.
Without this change english texts are read with a german accent if
the OS language is set to German and the app language to English.

Users will have to set QLocale::setDefault to their desired app
language in order for this behavior to work.

On macOS VoiceOver does not evaluate this property and texts are
always read with the OS language.

Pick-to: 6.8 6.10
Change-Id: I469652ff44f9d788ce6e6bdfa6c34d1570f2e439
---
 src/plugins/platforms/ios/quiaccessibilityelement.mm | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git x/qtbase/src/plugins/platforms/ios/quiaccessibilityelement.mm y/qtbase/src/plugins/platforms/ios/quiaccessibilityelement.mm
index 33321a626efed321d3cb5cacc17e6bce452233e9..12f519eaa9ec846319854b4beb9eb5d9895db6a7 100644
--- x/qtbase/src/plugins/platforms/ios/quiaccessibilityelement.mm
+++ y/qtbase/src/plugins/platforms/ios/quiaccessibilityelement.mm
@@ -259,6 +259,12 @@ QT_NAMESPACE_ALIAS_OBJC_CLASS(QMacAccessibilityElement);
     return NO;
 }
 
+- (NSString*)accessibilityLanguage
+{
+    const QLocale locale;
+    return locale.bcp47Name().toNSString();
+}
+
 @end
 
 #endif
